src/opm/output/eclipse/RegionCache.cpp

```diff
@@ -40,12 +40,12 @@ RegionCache::RegionCache(const std::set<std::string>& fip_regions, const FieldPr
                 continue;
 
             for (const auto& c : connections) {
-                if (grid.cellActive(c.getI(), c.getJ(), c.getK())) {
-                    size_t active_index = grid.activeIndex(c.getI(), c.getJ(), c.getK());
+                if (grid.cellActive(c.global_index())) {
+                    size_t active_index = grid.activeIndex(c.global_index());
                     int region_id = fip_region[active_index];
                     auto key = std::make_pair(fip_name, region_id);
                     auto& well_index_list = this->connection_map[ key ];
-                    well_index_list.push_back( { well.name() , active_index } );
+                    well_index_list.emplace_back(well.name(), c.global_index());
```

Comment:
I looked at the consequence of this change in Summary.cpp, and the most immediately affected place is on line 1485 of that file:
```
        double rate = args.wells.get( pair.first , pair.second , phase ) * eff_fac;
```
where in the above, `pair.second` will change from being an active cell index to a logical cartesian cell index. Looking at the implementation of that `get()` function in opm/output/data/Wells.hpp I see this statement:
```
const auto& connection = std::find_if( well.connections.begin() ,
                                       well.connections.end() ,
                                       [=]( const Connection& c ) {
                                            return c.index == connection_grid_index; });
```
... so is a Connection::index meant to be active or logical cartesian? Turns out the answer is logical cartesian, so I think this is indeed a bug fix! Well found.